lib/Pipeline.js

```diff
@@ -132,7 +132,9 @@ class Pipeline {
     }
 
     this.final = (callback) => {
-      this.firstStream.end(callback)
+      finished(this.lastStream, callback)
+
+      this.firstStream.end()
```

Comment:
And why is it not `this.lastStream.end()`?

Reply:
`.end()` triggers the `.final()` method of the `Writable` interface. That is propagated from the start to the end of a chain of streams connected with `.pipe()`.